### PR TITLE
collection view: adds stubbed layout

### PR DESCRIPTION
**Pull request by kendallstrautman**
## "Ready for Review" checklist

- [X] The Vercel preview link has been added to this PR's description
- [X] The Asana task has been added to this PR's description
- [X] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [ ] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [X] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-git-kscollection-view-layout-hashicorp.vercel.app/vault/tutorials/adp) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1202095033465927) 🎟️

## What

This PR adds the basic sidebar / sidecar layout to the collection view. This PR isn't meant to implement the collection sidebar to spec, but just getting things in place so that style refinement is easier. 

## Why

Setting things up so we can just focus on component development for this view!

## How

I adjusted the `server.js` file to organize the sidebar sections according to the collection categories. These props are passed as `layoutProps` to the view. 

## Testing

- [Visit this link](https://dev-portal-git-kscollection-view-layout-hashicorp.vercel.app/vault/tutorials/adp) and double check that the collection list matches that of [Learn today. ](https://learn.hashicorp.com/collections/vault/adp)
- Test that the sidebar links work
- Test that the breadcrumb links work
- Test that the sidecar links work. 

## Anything else?

The sidecar links in [the designs](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Dev-Portal?node-id=7053%3A54950) link to a section called 'tutorials'. In this PR I added a header with an id for this, but we'll need to think of how we link to this section without the header and still have the sidecar track properly. It seems the sidecar tracking is linked to the heading element specifically, but I could be wrong!

### CODE REVIEW

File: src/views/tutorial-view/server.ts [hunk 1]
```diff
@@ -52,14 +52,14 @@ export async function getTutorialPageProps(
   const layoutProps = {
     headings,
     breadcrumbLinks: getTutorialsBreadcrumb({
-      product: { name: product.name, slug: product.slug },
+      product: { name: product.name, filename: product.slug },
```
Comment (kendallstrautman) [inline src/views/tutorial-view/server.ts:55]: Just a rename here to make it more semantic 👍

File: src/views/product-tutorials-view/helpers.ts [hunk 2]
```diff
@@ -29,3 +35,10 @@ function sortAlphabetically(
     return 0
   }
 }
+
+export function filterAndSortCollections(
```
Comment (kendallstrautman) [inline src/views/product-tutorials-view/helpers.ts:39]: Separated out the sorting from filtering, to be more functional for this use case.

File: src/views/collection-view/components/collection-view-sidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+} from 'components/tutorials-sidebar'
+import { LearnProductData } from 'types/products'
+import { CollectionCategorySidebarSection } from '../helpers'
+
+interface CollectionViewSidebarProps {
+  product: LearnProductData
+  sections: CollectionCategorySidebarSection[]
+}
+
+export default function CollectionViewSidebar({
```
[the remark this answers is not in the document]
Comment (zchsh) [reply src/views/collection-view/components/collection-view-sidebar.tsx:14]: I for one love smaller component files, even if it means have to traverse more files 😍👍 very on board with this!

File: src/views/collection-view/index.tsx [hunk 4]
```diff
@@ -1,22 +1,31 @@
...
+      sidebarSlot={
+        <CollectionViewSidebar
+          product={product}
+          sections={layoutProps.sidebarSections}
+        />
+      }
+    >
+      <h1 id={layoutProps.headings[0].slug}>{name}</h1>
       <p>{description}</p>
+      <h2 id={layoutProps.headings[1].slug}>Tutorials</h2>
```
[the remark this answers is not in the document]
Comment (zchsh) [reply src/views/collection-view/index.tsx:28]: oooh yeah good callout 🤔

**Review by zchsh — APPROVED** [review]
✅ Looks great to me:

✓ - collections sidebar in preview matches Learn currently
✓ - sidebar & breadcrumb links work
✓ - sidecar links work (in the context of this PR)

re: sidecar links, I think this is outside the scope of this PR, but I am seeing some weird behaviour - possibly related to the amount of available space & only having two headings?

CleanShot 2022-04-13 at 14 49 31

Have opened [a task to investigate](https://app.asana.com/0/1202097197789424/1202125773561669/f) this cause I think it's outside the scope of this PR 👍

File: src/views/collection-view/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (ashleemboyer) [reply src/views/collection-view/index.tsx:28]: Do we _need_ to have this section be linkable? The `Sidecar` doesn't seem very useful in this view. The list of tutorials appears pretty close to the top of the page.

**Review by ashleemboyer — CHANGES_REQUESTED** [review]
Not sure if this is related to this PR or not, but the view that renders the following page is missing a back to link at the top of the sidebar: https://dev-portal-git-kscollection-view-layout-hashicorp.vercel.app/vault/tutorials/secrets-management/static-secrets.

File: src/views/collection-view/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (zchsh) [reply src/views/collection-view/index.tsx:28]: ☝️ +1, I think this is a great question for design. I raised it with Brayden, I think the intent here is that later there may be additional sections on the page? I think I remember it being mentioned that removing the `Sidecar` would create a strange whitespace balance as well.

Personally I'd be on board with removing it until we have more substance to put it in it; the whitespace balance drawback might also carry the benefit of better focus on the page (fewer elements).

[the remark this answers is not in the document]
Comment (kendallstrautman) [reply src/views/collection-view/index.tsx:28]: Yeah let's check in with design about this. I think for this view there wouldn't be extra sections and I agree it doesn't seem _super_ useful. Maybe until we get clarity we just have it there?

File: src/views/tutorial-view/server.ts (hunk 1 above)
Comment (kendallstrautman) [reply src/views/tutorial-view/server.ts:55]: That's fair, I just wanted the interface to be consistent. If its a blocker I can change it, but it seems pretty minor to me 🤷‍♀️

**Comment by kendallstrautman** [comment]
@ashleemboyer the link you shared that shows a different sidebar layout (without the back to link)) is a tutorial view, so this PR wouldn't affect that. I think @zchsh is working on getting the tutorial sidebar up to date.

File: src/views/collection-view/components/collection-view-sidebar.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  product,
+  sections,
+}: CollectionViewSidebarProps) {
+  return (
+    <TutorialsSidebar
+      title={`${product.name} Tutorials`}
```
[the remark this answers is not in the document]
Comment (kendallstrautman) [reply src/views/collection-view/components/collection-view-sidebar.tsx:20]: Cool, I was referencing the template which could be out of date. Will update!

File: src/pages/vault/tutorials/[collectionSlug].tsx [hunk 6]
```diff
@@ -7,9 +7,8 @@ import {
   getCollectionPaths,
   CollectionPageProps,
 } from 'views/collection-view/server'
-import BaseLayout from 'layouts/base-new'
 
-export function VaultCollectionPage(
+export default function VaultCollectionPage(
```
Comment (ashleemboyer) [inline src/pages/vault/tutorials/[collectionSlug].tsx:11]: @kendallstrautman what I was suggesting here is that this (and the one for Waypoint):

```tsx
export default function VaultCollectionPage(
  props: CollectionPageProps
): React.ReactElement {
  return <CollectionView {...props} />
}
```

get replaced with this:

```tsx
export default CollectionView
```

Which is what a few product landing pages do with `ProductLandingView`:

- [Vault](https://github.com/hashicorp/dev-portal/blob/main/src/pages/vault/index.tsx#L16)
- [Waypoint](https://github.com/hashicorp/dev-portal/blob/main/src/pages/waypoint/index.tsx#L16)

**Review by ashleemboyer — APPROVED** [review]
✨

**Pull request merged**